js/search.js

```diff
@@ -962,11 +961,38 @@
 				if (id === 'hiddenpowergrass') isViable = (moves.indexOf('energyball') < 0 && moves.indexOf('gigadrain') < 0);
 				if (id === 'hiddenpowerice') isViable = (moves.indexOf('icebeam') < 0 && template.id !== 'xerneas');
 				if (id === 'icywind') isViable = (toId(set.species).substr(0, 6) === 'keldeo');
+				if (id === 'leechlife') isViable = (this.gen > 6);
 				if (id === 'reflecttype') isViable = (toId(set.species) in {latias:1, starmie:1});
 				if (id === 'selfdestruct') isViable = (this.gen < 5 && moves.indexOf('explosion') < 0);
 				if (id === 'skyattack') isViable = (toId(set.species) === 'hawlucha');
 				if (id === 'stunspore') isViable = (moves.indexOf('thunderwave') < 0);
-				if (id === 'toxic') isViable = (this.gen > 1);
+				if (this.gen === 1) {
+					//Usually viable for Gen 1
+					if (id === 'acidarmor' || id === 'amnesia' || id === 'barrier' || id === 'bind' || id === 'clamp' || id === 'confuseray' || id === 'counter' || id === 'firespin' || id === 'hyperbeam' || id === 'mirrormove' || id === 'pinmissile' || id === 'razorleaf' || id === 'sing' || id === 'slash' || id === 'sludge' || id === 'twineedle' || id === 'wrap') isViable = true;
+
+					//Usually not viable for Gen 1
+					if (id === 'disable' || id === 'firepunch' || id === 'icepunch' || id === 'jumpkick' || id === 'leechseed' || id === 'quickattack' || id === 'roar' || id === 'thunder' || id === 'thunderpunch' || id === 'toxic' || id === 'triattack' || id === 'whirlwind') isViable = false;
+
+					//Viable only when certain moves aren't present
+					if (id === 'bubblebeam') isViable = (moves.indexOf('surf') < 0 && moves.indexOf('blizzard') < 0);
+					if (id === 'doubleedge') isViable = (moves.indexOf('bodyslam'));
+					if (id === 'hypnosis') isViable = (moves.indexOf('sleeppowder') < 0);
+					if (id === 'megadrain') isViable = (moves.indexOf('razorleaf') < 0 && moves.indexOf('surf') < 0);
+					if (id === 'megakick') isViable = (moves.indexOf('hyperbeam') < 0);
+					if (id === 'reflect') isViable = (moves.indexOf('barrier') < 0 && moves.indexOf('acidarmor') < 0);
+
+					//Viable for specific types
+					if (id === 'substitute') isViable = (template.types[0] === 'Ground' || template.types[1] === 'Ground');
+
+					//Viable for specific Pokemon
+					if (id === 'doublekick') isViable = (toId(set.species) === 'jolteon');
+					if (id === 'lightscreen') isViable = (toId(set.species) === 'zapdos');
+					if (id === 'meditate') isViable = (toId(set.species) === 'hitmonlee');
+
+					//Complicated viability
+					if (id === 'fireblast' || id === 'flamethrower') isViable = (template.types[0] === 'Fire' || template.types[1] === 'Fire' || toId(set.species) === 'aerodactyl');
```

Comment:
If you're adamant about this, I can remove the check, but generally Fire Blast and Flamethrower are undesirable in RBY due to inflicting burn and only rarely is useful for coverage. For the most part only Fire-types, Aerodactyl, and maybe Machamp want this.